### ISSUE TITLE
Breaking changes in DisplayManager in 1.5

### ISSUE THREAD

**Issue by ns8482e** [OrchardCMS/OrchardCore#12678]
### Describe the bug

I am using `BuildEditor` of `DisplayManager` to create a shape, It's working well, but when upgraded to 1.5 preview feed started getting following error.

![image](https://user-images.githubusercontent.com/23270244/197192034-343f0ab7-58d7-4239-bc5c-3ab80d622567.png)

**Comment by hishamco** [comment]
Could we add it to `1.5` milestone and handle it before the release?

**Comment by hishamco** [comment]
@ns8482e could you please add docs for this?

**Comment by hishamco** [comment]
@ns8482e could you please add docs for this?

**Comment by hishamco** [comment]
@ns8482e could you please add docs for this?

**Comment by ns8482e** [comment]
@hishamco it should be backed compatible, breaking the module

**Comment by hishamco** [comment]
@MikeAlhayek is not backward compatible?

**Comment by ns8482e** [comment]
@hishamco No it’s not, module written on 1.4 can find the method as Paramus of API has changed and now it’s complaining about method not found exception

**Comment by hishamco** [comment]
Hope if you can fix this before `1.5` is release, as I expect it will be ASAP

**Comment by MikeAlhayek** [comment]
@ns8482e your module depends on v1.3 so you can add dependency to your module for OC <=1.4. then create another package that depends on >=1.5. There is is really no to do in OC for this.

The issue is that your project is marked as compatible with 1.3+ when it's not.

**Comment by ns8482e** [comment]
@MikeAlhayek It's not just about my modules, It about how how OC Framework handles breaking changes gracefully in minor version upgrades.

**Comment by ns8482e** [comment]
> project is marked as compatible with 1.3+ when it's not

:) Can't predict that OC 1.5 will introduce breaking changes when released module 1.3 released

**Comment by ns8482e** [comment]
@hishamco I guess this can't be fixed as @MikeAlhayek suggests -requires new module to be recompiled

**Comment by hishamco** [comment]
Niraj please do the right fix, then you can request @jtkech

**Comment by ns8482e** [comment]
It's compiler - It doesn't allow me to add overload method as it thinks ambiguous.

@jtkech  original interface was  as below 

```cs
Task<IShape> BuildEditorAsync(TModel model, IUpdateModel updater, bool isNew, string groupId = "");
Task<IShape> UpdateEditorAsync(TModel model, IUpdateModel updater, bool isNew, string groupId = "");
```
That changed to following in 1.5-preview

```cs
 Task<IShape> BuildEditorAsync(TModel model, IUpdateModel updater, bool isNew, string groupId = "", string htmlPrefix = "");
 Task<IShape> UpdateEditorAsync(TModel model, IUpdateModel updater, bool isNew, string groupId = "", string htmlPrefix = "");
```

So all calls from module complied and released before 1.4 expects a method that has four params that results in method not found in when upgraded OC to 1.5-preview. 

Here in new method newly added default parameter `string htmlPrefix = ""` is not backward compatible and adding overload like `Task<IShape> BuildEditorAsync(TModel model, IUpdateModel updater, bool isNew, string groupId = "");` results in ambiguous compile error.

**Comment by ns8482e** [comment]
@hishamco @jtkech figured way fix it by keeping `htmlPrefix` without default value
